extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-workspace-service-impl.ts

```diff
@@ -0,0 +1,173 @@
+  }
+
+  public async getAll(userToken?: string): Promise<Workspace[]> {
+    throw new Error('workspaceService.getAll() not supported');
+  }
+
+  public async getAllByNamespace(namespace: string, userToken?: string): Promise<Workspace[]> {
+    throw new Error(`workspaceService.getAllByNamespace(${namespace}) not supported`);
+  }
+
+  public async updateWorkspace(workspaceId: string, workspace: Workspace): Promise<Workspace> {
+    throw new Error(`workspaceService.updateWorkspace(${workspaceId}) not supported`);
+  }
+
+  public async updateWorkspaceActivity(): Promise<void> {
+    throw new Error('workpsaceService.updateWorkspaceActivity() not supported');
+  }
+
+  public async stop(): Promise<void> {
+    // stopping the workspace is changing the state to false
+    throw new Error('workpsaceService.stop() not supported');
+  }
+
+  public async getWorkspaceSettings(): Promise<WorkspaceSettings> {
+    console.log('workspaceService.getWorkspaceSettings() not supported');
+    return {};
+  }
+
+  public async getCurrentWorkspacesContainers(): Promise<{ [key: string]: Container }> {
+    const result: { [key: string]: Container } = {};
+    try {
+      const workspace = await this.currentWorkspace();
+      const containers = workspace.runtime!.machines;
+      if (containers) {
+        for (const containerName of Object.keys(containers)) {
+          const container: Container = { name: containerName, ...containers[containerName] };
+          container.name = containerName;
+          if (container) {
+            result[containerName] = container;
+          }
+        }
+      }
+    } catch (e) {
+      throw new Error(`Unable to get workspace containers. Cause: ${e}`);
+    }
+    return result;
+  }
+
+  public async findUniqueEndpointByAttribute(attributeName: string, attributeValue: string): Promise<Endpoint> {
+    const containers = await this.getCurrentWorkspacesContainers();
+    try {
+      if (containers) {
+        for (const containerName of Object.keys(containers)) {
+          const servers = containers[containerName].servers;
+          if (servers) {
+            for (const serverName of Object.keys(servers)) {
+              const server = servers[serverName];
+              if (server && server.attributes && server.attributes[attributeName] === attributeValue) {
+                return server;
+              }
+            }
+          }
+        }
+      }
+      return Promise.reject(`Server by attributes '${attributeName}'='${attributeValue}' was not found.`);
+    } catch (e) {
+      return Promise.reject(`Unable to get workspace servers. Cause: ${e}`);
+    }
+  }
+
+  public async getContainerList(): Promise<Container[]> {
+    const containers: Container[] = [];
+    try {
+      const workspace = await this.currentWorkspace();
+
+      if (workspace.runtime && workspace.runtime.machines) {
+        const machines = workspace.runtime.machines;
```

Comment:
`workspace.runtime` is deprecated now
shouldn't the `EndpointService` be used here?

Reply:
yes something is wrong 👍